numpy/typing/_callback.py

```diff
@@ -0,0 +1,132 @@
+"""A module with various callback protocols."""
```

Comment:
Do you mean "callable"? It looks to me like the goal here is to provide common overloaded function / method signatures, if that helps you clarify the comment.

Reply:
The [mypy docs](https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols) refer to these type of protocols as callback protocols, 
presumably because (them being classes) protocols are always callable objects.

I expanded the module-level docstring a bit in https://github.com/numpy/numpy/pull/17273/commits/8608127f59518411ab0e5e8e0c5059ac839b0c0a in order to provide further clarity on the subject.

Reply:
I disagree - the mypy docs are showing how this type of protocol can be _used_ for callbacks. But in the numpy stubs here, these _aren't_ callabacks. A callback is a callable that is a function argument, which these are not.

> presumably because (them being classes) protocols are always callable objects.

You're conflating the class itself, (which is callable) vs class instances (which are only callable if they define `__call__`).

Reply:
> A callback is a callable that is a function argument, which these are not.

I wasn't aware of this distinction; I see your point.
Fixed in https://github.com/numpy/numpy/pull/17273/commits/58c81b38b293370edbeee4552d12951a5792ca1e.